pipeline/deploy-staging.sh

```diff
@@ -34,12 +68,32 @@ then
 	git add . || exit 1
 	git config user.email "svc.bot-cns@vmware.com" || exit 1
 	git config user.name "svc.bot-cns" || exit 1
-	git commit -m "Pipeline updated staging/patch.yaml with images $VSPHERE_CSI_CONTROLLER_IMAGE and $VSPHERE_SYNCER_IMAGE" || exit 1
+	git commit -m "Pipeline updated manifest files with images $VSPHERE_CSI_CONTROLLER_IMAGE and $VSPHERE_SYNCER_IMAGE" || exit 1
 	git push origin main || exit 1
 else
 	echo "No code changes pushed to the staging repo."
 fi
 
-# TODO: Add code to wait for the CD infra to update the CSI in the staging environment.
+while true
+do
+	tmp_file=/tmp/$$
+	kubectl -n vmware-system-csi get deployment vsphere-csi-controller -o=jsonpath="{'\n'}{range .spec.template.spec.containers[*]}{.image}{'\n'}{end}" | tee $tmp_file
+	if grep "$VSPHERE_CSI_CONTROLLER_IMAGE" $tmp_file;
+	then
+		echo "CSI deployment is patched."
+		break
+	fi
+	echo "CSI deployment not yet patched. Sleeping for 1 second..."
+	sleep 1
+done
+
+echo "Sleeping for 1 min for WCP to reconcile the CSI patch.."
```

Comment:
if we do not want to wait for WCP to reconcile the CSI patch for 1 minute and then later wait for ready replica to be 3, i think we can do following

```
kubectl set image -n vmware-system-csi deployment/vsphere-csi-controller vsphere-csi-controller=new-csi-controller-image
kubectl rollout restart -n vmware-system-csi deployment/vsphere-csi-controller --timeout=120s
```

Reply:
No, we don't want to manually set the image and rollout. The idea is to use GitOps(via Argo CD) to deploy the CSI manifests to the staging environment. Note that the dev's commits may contain changes to the CSI manifest files as well. The wait here is needed to give some time for WCP to start the reconciliation process to process the CSI patch. Without this 1 min sleep, the next step (wait for the readyReplica=3) may be reported incorrectly based on the previous CSI manifest definition.